Variable "pydantic_core.core_schema.FieldValidationInfo" is not valid as a type

The deprecation of `pydantic_core.core_schema.FieldValidationInfo` in v2.10.0 makes mypy unhappy:

```python
from pydantic_core.core_schema import FieldValidationInfo

def f(x: FieldValidationInfo):
    ...
```

```console
$ mypy test.py
test.py:3: error: Variable "pydantic_core.core_schema.FieldValidationInfo" is not valid as a type  [valid-type]
test.py:3: note: See https://mypy.readthedocs.io/en/stable/common_issues.html#variables-vs-type-aliases
Found 1 error in 1 file (checked 1 source file)
```

No error occurs when using v2.9.0.

This behavior turns the deprecation into a breaking change (at least regarding type-checking). As a result, we can't update from Pydantic v2.3 to v2.4 as `pydantic-core` is a pinned dependency and bumped to v2.10.

--- a/python/pydantic_core/core_schema.py
+++ b/python/pydantic_core/core_schema.py
@@ -3909,6 +3909,9 @@ _deprecated_import_lookup = {
     'FieldWrapValidatorFunction': WithInfoWrapValidatorFunction,
 }
 
+if TYPE_CHECKING:
+    FieldValidationInfo = ValidationInfo
+
 
 def __getattr__(attr_name: str) -> object:
     new_attr = _deprecated_import_lookup.get(attr_name)

--- a/tests/test_typing.py
+++ b/tests/test_typing.py
@@ -23,6 +23,10 @@ def foo(bar: str) -> None:
     ...
 
 
+def validator_deprecated(value: Any, info: core_schema.FieldValidationInfo) -> None:
+    ...
+
+
 def validator(value: Any, info: core_schema.ValidationInfo) -> None:
     ...
 

